lib/local-links-manager/import/csv_downloader.rb

```diff
@@ -5,12 +5,19 @@ class Error < RuntimeError; end
   class DownloadError < Error; end
   class MalformedCSVError < Error; end
 
-  def initialize(csv_url)
+  def initialize(csv_url, header_conversions = {})
     @csv_url = csv_url
+
+    if header_conversions
+      CSV::HeaderConverters[:name_converter] = name_converter(header_conversions)
```

Comment:
Could we avoid setting a named converter entirely and use the lambda directly:

```
def initialize(csv_url, header_conversions = {})
  @csv_url = csv_url
  @header_conversions = header_conversions
end

def download
  CSV.parse(downloaded_csv, headers: true, header_converters: name_converter)
end

def name_converter
  lambda do |field|
    @header_conversions.detect { |k, _| k == field } ? @header_conversions[field] : field
  end
end
```

By setting it as a named converter in `CSV::HeaderConverters[:name_converter]` we expose that conversion globally.  Crucially, if one of our `CsvDownloader` instances does use header conversions (and so sets the `:name_converter` here) but the instance of `CsvDownloader` doesn't use header conversions (and so doesn't set the `:name_converter` here) then this 2nd instance would end up using the `:name_converter` set by the previous instance.  It'll get even more weird if we had multiple `CsvDownloader` instances running at the same time in threads - ultimately the last `:name_converter` set would be used by all instances.